drivers/gpio/gpio_sam0.c

```diff
@@ -299,86 +299,64 @@ static const struct gpio_driver_api gpio_sam0_api = {
 
 static int gpio_sam0_init(struct device *dev) { return 0; }
 
-/* Port A */
-#if DT_ATMEL_SAM0_GPIO_PORT_A_BASE_ADDRESS
-
-static const struct gpio_sam0_config gpio_sam0_config_0 = {
-	.common = {
-		.port_pin_mask = GPIO_PORT_PIN_MASK_FROM_NGPIOS(DT_INST_0_ATMEL_SAM0_GPIO_NGPIOS),
-	},
-	.regs = (PortGroup *)DT_ATMEL_SAM0_GPIO_PORT_A_BASE_ADDRESS,
 #ifdef CONFIG_SAM0_EIC
```

Comment:
To avoid defining the `GPIO_SAM0_DEVICE_CONFIG` macro twice we could do 
```
#ifdef CONFIG_SAM0_EIC
#define GPIO_SAM0_EIC_ID .id = n,
#else
#define GPIO_SAM0_EIC_ID
#endif
```
and then use `GPIO_SAM0_EIC_ID` in `GPIO_SAM0_DEVICE_CONFIG`.

Reply:
Done!